evm/test/ConversionRate_test.js

```diff
@@ -566,4 +567,67 @@ contract('ConverstionRate', () => {
       })
     }
   })
+
+  describe('#cancelRequest', () => {
+    let request
+
+    beforeEach(async () => {
+      rate = await h.deploy(
+        SOURCE_PATH,
+        link.address,
+        basePayment,
+        1,
+        [oc1.address],
+        [jobId1]
+      )
+
+      await link.transfer(rate.address, basePayment)
+
+      assertBigNum(basePayment, await link.balanceOf.call(rate.address))
+      assertBigNum(0, await link.balanceOf.call(oc1.address))
+
+      const requestTx = await rate.requestRateUpdate()
+      request = h.decodeRunRequest(requestTx.receipt.rawLogs[3])
+
+      assertBigNum(0, await link.balanceOf.call(rate.address))
+      assertBigNum(basePayment, await link.balanceOf.call(oc1.address))
+
+      await h.increaseTime5Minutes() // wait for request to expire
+    })
+
+    context('when a later answer has been provided', () => {
+      beforeEach(async () => {
+        await link.transfer(rate.address, basePayment)
+        const requestTx2 = await rate.requestRateUpdate()
+        const request2 = h.decodeRunRequest(requestTx2.receipt.rawLogs[3])
+        await h.fulfillOracleRequest(oc1, request2, 17)
+      })
+
+      it('gets the LINK deposited back from the oracle', async () => {
+        await rate.cancelRequest(
+          request.id,
+          request.payment,
+          request.expiration
+        )
+
+        assertBigNum(basePayment, await link.balanceOf.call(rate.address))
+        assertBigNum(basePayment, await link.balanceOf.call(oc1.address))
```

Comment:
Should one of these asserts be testing for 0? Seems like the amount of link has doubled, relative to the situation in the `beforeEach`.

Reply:
On line 600 we send in another base payment, so that we can complete a later answer and withdraw the first deposit. Could probably use a precondition check in the `beforeEach` or at the beginning of this test to show that we are dealing with two deposits and not just one.